### PR TITLE
Add C++ nn::Identity

### PR DESCRIPTION
**Pull request by jon-tow**
**Summary**: 
Adds `torch::nn::Identity` module support for the C++ API.

**Issue**: #25883  

**Reviewer**: @yf225

### CODE REVIEW

**Review by yf225 — APPROVED** [review]
@jon-tow Thanks a lot and it looks awesome! I will land it today 😄

**Comment by yf225** [comment]
@jon-tow Please let me know if by any slight chance you would be interested in contributing any other layers in https://github.com/pytorch/pytorch/issues/25883. Really appreciated your help and it makes the C++ API much more awesome :)

**Comment by jon-tow** [comment]
@yf225 Sure! I would love to. I was looking to add some loss functions but noticed a discussion on handling reduction enums, so I figured I'd wait until that issue clears up to contribute again. 
I'll get started on the layers first thing in the morning (EST) :).

**Comment by yf225** [comment]
@jon-tow Thanks so much again for your help! And please feel free to use `enum class` for enums for now - after https://github.com/pytorch/pytorch/pull/26837 we will have more clarity on the best practices for enum usage, and we can easily update the PRs accordingly :) Thanks again!

**Comment by yf225** [comment]
@jon-tow Please let me know the layers you would like to write if any, and I will update https://github.com/pytorch/pytorch/issues/25883 accordingly. Thanks!
